zulipterminal/core.py

```diff
@@ -492,6 +492,37 @@ def show_media_confirmation_popup(
             self, question, callback, location="center"
         )
 
+    def extract_keywords_from_search_query(
+        self, query: str
+    ) -> Tuple[str, List[List[str]]]:
+        """
+        This function checks if any prefixes are present in the search string or not,
+        it returns a newly created narrow as per the prefixes and the query without the prefixes.
+        """
+        # FIXME: Currently, the function does not support pm_with keyword in search text
+        stream_prefix_present_flag = False
+        final_search_text = ""
+        new_narrow = []
+        for word in query.split(" "):
+            if word.startswith("stream:"):
+                new_narrow = [
+                    ["stream", word.split(":")[1].replace("+", " ").replace("%2B", "+")]
+                ]
+                stream_prefix_present_flag = True
+            elif word.startswith("topic:") and stream_prefix_present_flag:
+                new_narrow.append(
+                    ["topic", word.split(":")[1].replace("+", " ").replace("%2B", "+")]
```

Comment:
It would be good to see various of these replacement operations tested - other than testing the code, it also demonstrates the input->output one might expect, on reading the tests.